modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java

```diff
@@ -621,6 +621,8 @@ Object encodeLiteral(Object literal, Class<?> targetType) {
             }
             // by default, return string as is
             return literal;
+        } else if (literal instanceof Number) {
+            return literal;
```

Comment:
Agreed, checked the driver docs and for now I think Int, Long Double only is the best option.

@jodygarnett you mention, 'throw an appropriate unsupported exception for the case you miss.'  That was my initial thought as well, but didn't fully understand the original intention of this, which made me leery of that change:

https://github.com/newmanw/geotools/blob/9c0f61669551b6e04b6a7ab6c81b2cd6efdbe482/modules/unsupported/mongodb/src/main/java/org/geotools/data/mongodb/FilterToMongo.java#l626-L628 

which was the original implementation.  Thoughts?